app/src/org/commcare/tasks/ManageKeyRecordTask.java

```diff
@@ -150,7 +150,7 @@ protected void deliverError(R receiver, Exception e) {
 
     protected void keysReadyForSync(R receiver) {
         // TODO: we only wanna do this on the _first_ try. Not subsequent ones (IE: On return from startDataPull)
-        receiver.startDataPull(this.dataPullMode);
+        receiver.startDataPull(this.dataPullMode, password);

```

Comment:
This might be a silly question, but does this mean that in subsequent logins using pin, this password will be sent as null?  cuz [here](https://github.com/dimagi/commcare-android/pull/2370/files#diff-4da5bd220b747979dec7869dd4d3bc7fR92-R98) it's being set to null for pin login mode

Reply:
Password gets set from [here](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/tasks/ManageKeyRecordTask.java#L460) in pin mode, so it should not be null.

Reply:
gotcha